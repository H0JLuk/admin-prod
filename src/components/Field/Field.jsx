import React from 'react';
import PropTypes from 'prop-types';
import styles from './Field.module.css';

const Field = ({ label, value }) => {
    return (
        <p className={ styles.text }>
            <span className={ styles.bold }>{ label }</span>
            { value }
        </p>
    );
};

Field.propTypes = {
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired
};

export default Field;
