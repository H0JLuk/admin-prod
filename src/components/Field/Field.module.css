.text {
    font-family: Tahoma, sans-serif;
    font-size: 16px;
    margin-bottom: 3px;
}

.bold {
    font-weight: bold;
}
