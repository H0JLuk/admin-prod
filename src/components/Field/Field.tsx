import React from 'react';
import PropTypes from 'prop-types';
import styles from './Field.module.css';

type FieldProps = {
    label: string;
    value: string;
};

// TODO: Удалить после редизайна и удаления старого компонента страницы категорий
const Field: React.FC<FieldProps> = ({ label, value }) => (
    <p className={styles.text}>
        <span className={styles.bold}>{label}</span>
        {value}
    </p>
);

Field.propTypes = {
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired
};

export default Field;
