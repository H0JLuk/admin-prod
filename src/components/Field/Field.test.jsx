import React from 'react';
import { shallow } from 'enzyme';
import Field from './Field';

describe('<Field /> with no props', () => {
    console.error = jest.fn();
    const container = shallow(<Field />);

    it('should match the snapshot', () => {
        expect(container.html()).toMatchSnapshot();
    });

    it('check empty values', () => {
        expect(container.find('span').text()).toEqual('');
        expect(container.html()).toEqual('<p class="text"><span class="bold"></span></p>');
    });
});

describe('<Field /> with props', () => {
    const initialProps = {
        label: 'test',
        value: 'test_value',
    };
    const container = shallow(<Field { ...initialProps } />);

    it('should match the snapshot', () => {
        expect(container.html()).toMatchSnapshot();
    });

    it('check empty values', () => {
        expect(container.find('span').text()).toEqual(initialProps.label);
        expect(container.html()).toEqual(`<p class="text"><span class="bold">${initialProps.label}</span>${initialProps.value}</p>`);
    });
});
