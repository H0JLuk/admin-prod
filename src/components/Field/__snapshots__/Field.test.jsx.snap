// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Field /> with no props should match the snapshot 1`] = `"<p class=\\"text\\"><span class=\\"bold\\"></span></p>"`;

exports[`<Field /> with props should match the snapshot 1`] = `"<p class=\\"text\\"><span class=\\"bold\\">test</span>test_value</p>"`;
